x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/registry/ModelRegistry.java

```diff
@@ -935,6 +943,11 @@ static List<InferenceService.DefaultConfigId> taskTypeMatchedDefaults(
 
     @Override
     public void clusterChanged(ClusterChangedEvent event) {
+        // keep track of the last applied cluster state
+        synchronized (this) {
```

Comment:
Any blocking concerns using `synchronized` here?

Reply:
This is not strictly necessary but since it's "just" an assignment and a read with happens-before behaviour (volatile), I think we should be fine.